### PR TITLE
Add redis SSL parameter to RedisLockStore

### PR DESCRIPTION
**Pull request by ricwo**
**Proposed changes**:
- adds missing SSL parameter, analogous to https://github.com/RasaHQ/rasa/pull/4420

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/core/lock_store.py [hunk 1]
```diff
@@ -267,11 +267,12 @@ def __init__(
         port: int = 6379,
         db: int = 1,
         password: Optional[Text] = None,
+        ssl: bool = False,
```
Comment (wochinge) [inline rasa/core/lock_store.py]: We ended up calling it `use_ssl` for the tracker store

File: docs/api/lock-stores.rst [hunk 2]
```diff
@@ -66,3 +66,4 @@ RedisLockStore
     - ``db`` (default: ``0``): The number of your redis database
     - ``password`` (default: ``None``): Password used for authentication
       (``None`` equals no authentication)
+    - ``ssl`` (default: ``False``) Whether or not the communication is encrypted
```
Comment (wochinge) [inline docs/api/lock-stores.rst]:
```suggestion
    - ``ssl`` (default: ``False``): Whether or not the communication is encrypted
```

**Comment by ricwo** [comment]
@wochinge yeah it should - updated the base branch

**Comment by akelad** [comment]
i believe @tmbo took care of merging the 1.2.x branch into 1.4.x no?

**Comment by tmbo** [comment]
I did merge 1.3.x -> master.

I can't merge all 1.x.x branches before a release. If anyone merges and releases from an older branch they are responsible to merge it into master.

**Comment by tmbo** [comment]
I've cleaned this up and will tag a 1.4.1

**Pull request merged**